halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/ApacheSpinnakerProfileFactory.java

```diff
@@ -18,16 +18,23 @@
 package com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile;
 
 import com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentConfiguration;
+import com.netflix.spinnaker.halyard.config.model.v1.security.ApacheSsl;
+import com.netflix.spinnaker.halyard.config.model.v1.security.UiSecurity;
 import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerArtifact;
 import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerRuntimeSettings;
 import org.springframework.stereotype.Component;
 
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 @Component
 public class ApacheSpinnakerProfileFactory extends TemplateBackedProfileFactory {
   private static String SPINNAKER_TEMPLATE = "<VirtualHost {%deck-host%}:{%deck-port%}>\n"
+      + "  SSLEngine {%ssl-engine%}\n"
+      + "  SSLCertificateFile \"{%cert-file%}\"\n"
+      + "  SSLCertificateKeyFile \"{%key-file%}\"\n"
+      + "\n"
```

Comment:
Consider using `String.join` in order to reduce a bunch of `\n`s everywhere:

```
String s = String.join("\n"
         , "It was the best of times, it was the worst of times,"
         , "it was the age of wisdom, it was the age of foolishness,"
         , "it was the epoch of belief, it was the epoch of incredulity,"
         , "it was the season of Light, it was the season of Darkness,"
         , "it was the spring of hope, it was the winter of despair,"
         , "we had everything before us, we had nothing before us"
);
```
Source: http://stackoverflow.com/questions/878573/java-multiline-string